app/workers/certificates.php

```diff
@@ -26,169 +28,224 @@ public function init(): void
+                throw new ExceptionCertificate('Failed to rename certificate cert.pem: '.\json_encode($stdout), true);
+            }
+    
+            if (!@\rename('/etc/letsencrypt/live/' . $domain->get() . '/chain.pem', APP_STORAGE_CERTIFICATES . '/' . $domain->get() . '/chain.pem')) {
+                throw new ExceptionCertificate('Failed to rename certificate chain.pem: ' . \json_encode($stdout), true);
+            }
+    
+            if (!@\rename('/etc/letsencrypt/live/' . $domain->get() . '/fullchain.pem', APP_STORAGE_CERTIFICATES . '/' . $domain->get() . '/fullchain.pem')) {
+                throw new ExceptionCertificate('Failed to rename certificate fullchain.pem: ' . \json_encode($stdout), true);
+            }
+    
+            if (!@\rename('/etc/letsencrypt/live/' . $domain->get() . '/privkey.pem', APP_STORAGE_CERTIFICATES . '/' . $domain->get() . '/privkey.pem')) {
+                throw new ExceptionCertificate('Failed to rename certificate privkey.pem: ' . \json_encode($stdout), true);
+            }
 
-            $certificate = $dbForConsole->updateDocument('domains', $certificate->getId(), $certificate);
-
-            if(!$certificate) {
-                throw new Exception('Failed saving domain to DB');
+            // This multi-line syntax helps IDE
+            $config =
+                "tls:" .
+                "  certificates:" . 
+                "    - certFile: /storage/certificates/{$domain->get()}/fullchain.pem" . 
+                "      keyFile: /storage/certificates/{$domain->get()}/privkey.pem";
+            
+            // Save configuration into Traefik using our new cert files
+            if (!\file_put_contents(APP_STORAGE_CONFIG . '/' . $domain->get() . '.yml', $config)) {
+                throw new ExceptionCertificate('Failed to save Traefik configuration.', true);
             }
-        }
 
-        $config =
-"tls:
-  certificates:
-    - certFile: /storage/certificates/{$domain->get()}/fullchain.pem
-      keyFile: /storage/certificates/{$domain->get()}/privkey.pem";
+            // Read new renew date from cert file
+            // TODO: This might not be required, we could calculate it. But this feels safer
+            $certPath = APP_STORAGE_CERTIFICATES . '/' . $domain->get() . '/cert.pem';
+            $certData = openssl_x509_parse(file_get_contents($certPath));
+            $validTo = $certData['validTo_time_t'];
+            $expiryInAdvance = (60*60*24*30);
+            $certificate->setAttribute('renewDate', $validTo - $expiryInAdvance);
+
+            // All went well at this point 🥳
+            
+            // Reset attempts count for next renwal
+            $certificate->setAttribute('attempts', 0);
+
+            // Mark issue date
+            $certificate->setAttribute('issueDate', \time());
+        } catch(ExceptionCertificate $e) {
```

Comment:
We have introduced a proper error handler for all workers to avoid big try-catch blocks, let's use it.
https://github.com/appwrite/appwrite/blob/37c6fd75d8c72c385ed4f719466325beaf1a4d29/src/Appwrite/Resque/Worker.php#L146

Reply:
I dont think we can use it. Inside error(), we only have exception, not certificateID or other stuff. Should we pass these inside exception?

Reply:
Would it make sense to have it stored in a worker `$this->certificate` attribute?

Reply:
I don't think this is possible due to following reasons:

- Error handlers are registered "forever".. And there isn't a function in worker that would run only once.. So we need to implement unregister logic, and store error handlers under some key
- Error callbacks are stored on static variable. If we run 2 actions at the same time, 1 action will trigger errorCallback for second too. We would need to make Worker aware of some action ID and make `error()` not static
- Finally block would not be possible. We would need duplicate code, or one function at the top

A refactor of error handling in worker would be required before making this change.